api/configuration/v1alpha1/kongpluginbinding_types.go

```diff
@@ -96,8 +96,8 @@ type KongPluginBindingSpec struct {
 	Targets *KongPluginBindingTargets `json:"targets,omitempty"`
 
 	// ControlPlaneRef is a reference to a ControlPlane this KongPluginBinding is associated with.
-	// +optional
-	ControlPlaneRef *ControlPlaneRef `json:"controlPlaneRef,omitempty"`
+	// +kubebuilder:validation:Required
+	ControlPlaneRef *ControlPlaneRef `json:"controlPlaneRef"`
```

Comment:
This being `required` now I believe it should be changed to non pointer. AFAIK this is a well known convention in k8s, e.g. mentioned by (still unofficial, [kal](https://github.com/JoelSpeed/kal) linter, [related doc](https://docs.google.com/document/d/1O3ocYMfRtXXWG9sUT0tB3LUo5HbhBQKK62bohUjlLAc/edit?tab=t.0#heading=h.a9c74vwvzosc) that was brought up recently on SIG API machinery meeting in December: https://docs.google.com/document/d/1x9RNaaysyO0gXHIr1y50QFbiL1x8OWnk2v3XnrdkT5Y/edit?tab=t.0#heading=h.39diec6k8uaz)

Reply:
👍 Good call. Nice, https://github.com/JoelSpeed/kal looks like something that we should use. I created an issue for this: https://github.com/Kong/kubernetes-configuration/issues/246